internal/bgp/community/community.go

```diff
@@ -0,0 +1,141 @@
+	communityType       bgpCommunityType
+	globalAdministrator uint32
+	localDataPart1      uint32
+	localDataPart2      uint32
+}
+
+// String returns the string representation of this community. Legacy communities will be printed in new-format,
+// "<AS number>:<community value>". Large communities will be printed as
+// ("<global administrator>:<localdata part 1>:<localdata part 2>").
+func (b BGPCommunity) String() string {
+	if b.communityType == legacyBGPCommunity {
+		return legacyCommunityToString(b.globalAdministrator)
+	}
+	return fmt.Sprintf("%d:%d:%d", b.globalAdministrator, b.localDataPart1, b.localDataPart2)
+}
+
+func legacyCommunityToString(c uint32) string {
+	upperVal := c >> 16
+	lowerVal := c & 0xFFFF
+	return fmt.Sprintf("%d:%d", upperVal, lowerVal)
+}
+
+// LessThan makes 2 different BGPCommunity objects comparable. For the sake of comparison, legacy communities are
+// considered to be large communities in format <legacy community>:0:0 and thus can be compared with large communities.
+func (b BGPCommunity) LessThan(c BGPCommunity) bool {
+	return b.globalAdministrator < c.globalAdministrator ||
+		(b.globalAdministrator == c.globalAdministrator && (b.localDataPart1 < c.localDataPart1 ||
+			(b.localDataPart1 == c.localDataPart1 && b.localDataPart2 < c.localDataPart2)))
+}
+
+// IsLegacy returns true if this community is of type legacy.
+func (b BGPCommunity) IsLegacy() bool {
+	return b.communityType == legacyBGPCommunity
+}
+
+// IsLarge returns true if this community is of type large.
+func (b BGPCommunity) IsLarge() bool {
+	return b.communityType == largeBGPCommunity
+}
+
+// ToLegacy returns the uint32 representation of this community. If this community is not of type legacy, throw an
+// error.
+func (b BGPCommunity) ToLegacy() (uint32, error) {
+	if !b.IsLegacy() {
+		return 0, fmt.Errorf("community %s is not a legacy BGP Community", b)
+	}
+	return b.globalAdministrator, nil
+}
+
+// ToLarge returns the [3]uint32 representation of this community. If this community is not of type large, throw an
+// error.
+func (b BGPCommunity) ToLarge() ([]uint32, error) {
+	if !b.IsLarge() {
+		return nil, fmt.Errorf("community %s is not a large BGP Community", b)
+	}
+	return []uint32{b.globalAdministrator, b.localDataPart1, b.localDataPart2}, nil
+}
+
+// NewBGPCommunity parses the provided string and returns a newly created BGPCommunity object.
+// Strings are parsed according to Juniper style  syntax (https://www.juniper.net/documentation/us/en/software/\
+// junos/routing-policy/bgp/topics/concept/policy-bgp-communities-extended-communities-match-conditions-overview.html
+// Legacy communities are of format "<AS number>:<community value>".
+// Extended communities (support of which is yet to be implemented) are of format "<type>:<administrator>:<assigned-number>".
+// Large communities are of format large:<global administrator>:<localdata part 1>:<localdata part 2>.
+func NewBGPCommunity(c string) (BGPCommunity, error) {
```

Comment:
this can be just "New" as on the reader side it reads as community.New

Reply:
also, can we have this on top as it is the entry point

Reply:
done